modules/kn-service-update.adoc

```diff
@@ -0,0 +1,53 @@
+// Module included in the following assemblies:
+//
+// * serverless/knative_serving/serverless-applications.adoc
+// * serverless/cli_reference/kn-services-ref.adoc
+
+[id="kn-service-update_{context}"]
+= Updating serverless applications
+
+You can update a Knative service by using the `kn` CLI `kn service update` command.
```

Comment:
As mentioned in the description of `kn service apply`, we should add an explanation the difference between `update` and `apply` here (or over there), because that can be easily confused.

Reply:
Yep, sounds good. If you can supply input I'll add it.

Reply:
My take:

----
In contrast to the `kn service apply` command, you only provide the values that need to be changed. 
The Knative service needs to exist before it can be updated.
----

Thinking again, maybe we just skip extra blurb, as this might confuse people more than it helps, as this is the 'normal' way that I think people expect from an update. It's more like that `kn service apply` is special, not so much `kn service update`

Feel free to pick this up or leave it as is :)

Some background info (but feel free to ignore, but I couldn't resist :)

* `kn service apply` is used for a _declarative_ management of Knative services, i.e. you specify the target state and the backend controllers are responsible to move the current state to this target state ("reconciliation"). It's more or less equivalent to `kubectl apply -f ksvc.yaml`. This is especially useful for scripting and CI/CD systems where you are not sure about the current state of the service (does it exist ? or has it to be created ?)
* `kn service create` and `kn service update` is called _imperative_ management as the user imperatively tells the backend what to do ("create the service, then update the env, then update the labels, ...").

There is kind of a religious dispute among Kubernetes users which mode is better (most would probably say, the _declarative_ mode is preferred), so `kn supports` both ;-)

Reply:
Thanks for the extra context!
Yeah I think re this being the _usual_ way to do it, it's OK to leave as is and we have included the blurb in the declarative procedure instead.

Sort of unrelated questions, but
> more or less equivalent to kubectl apply -f ksvc.yaml

Does this mean that theoretically one could install `kn` and then use this to install Knative, i.e. by applying the serving and eventings CRDs? I realize it's not relevant for downstream since we use the operator but would be interesting for upstream if you could use `kn` to install and set up everything? Feel free to take this to Slack or emails instead of replying here 😂 

I think I'm in the declarative camp 😄 This was an interesting read, thank you.

Reply:
`kn` does not allow you to install Knative. So the command is actually `kn service apply -f ksvc.yaml` and `kn` verifies that `ksvc.yaml` indeed contains a service and will throw an error otherwise. So, no, kn can't be 'misused' for `kubectl apply` ;-)